### PR TITLE
WIP: Update docs for Spark 2.0 and Spark SQL

### PR DESCRIPTION
**Pull request by mdzimmerman**
Still largely a framework, but I put in suggested section heading and
comments with potential content.

Signed-off-by: Matthew Zimmerman <matt.zimmerman@ccri.com>

### CODE REVIEW

File: docs/user/spark.rst [hunk 1]
```diff
@@ -1,14 +1,193 @@
...
+other GeoTools ``DataStore``\ s, or files readable by the GeoMesa converter library.
+The library allows creation of Spark ``RDD``\ s and ``DataFrame``\ s, writing of
+Spark ``RDD``\ s and ``DataFrame``\ s to GeoMesa Accumulo and other GeoTools ``DataStore``\ s, and serialization of ``SimpleFeature``\ s using Kryo.
+
+The current version of GeoMesa Spark supports Apache Spark 2.0.
+
+Example
+-------
+
+.. short example?
```
Comment (jnh5y) [inline docs/user/spark.rst]: @Afterglow375

File: docs/user/spark.rst [hunk 2]
```diff
@@ -1,14 +1,193 @@
...
+
+Architecture
+------------
+
+The ``geomesa-spark-core`` module provides the core of GeoMesa's Spark support,
+through the ``GeoMesaSpark`` object...
+
+.. spark-sql
+.. include architecture diagram(s) from the Jupyter powerpoint?
```
Comment (jnh5y) [inline docs/user/spark.rst]: @mdzimmerman @jnh5y

File: docs/user/spark.rst [hunk 3]
```diff
@@ -1,14 +1,193 @@
...
+      <groupId>org.locationtech.geomesa</groupId>
+      <artifactId>geomesa-spark-geotools_2.11</artifactId>
+      // version, etc.
+    </dependency>
+
+.. introduction
+.. parameters/configuration
+.. don't use for Accumulo; use Accumulo provider above instead
+
+Spark SQL
```
Comment (jnh5y) [inline docs/user/spark.rst]: @Afterglow375

File: docs/user/spark.rst [hunk 4]
```diff
@@ -1,14 +1,193 @@
...
+---------
+
+.. introduction
+.. custom Spark types (Geometry, Point, Linestring, etc.)
+.. how certain queries are pushed down to the Accumulo/GeoTools layer
+.. broadcast and joins (and caveats thereof)
+
+Spatial Functions
```
Comment (jnh5y) [inline docs/user/spark.rst]: @mdzimmerman

File: docs/user/spark.rst [hunk 5]
```diff
@@ -1,14 +1,193 @@
...
 -----
 
-To use GeoMesa with Spark, the executors must know how to serialize and deserialize Simple Features. There are two ways
+.. how to create a new ``SparkSession``/``SparkContext``
+.. set up DS and work with them
+
+Jupyter
+-------
+
+.. setup: (Toree kernel, etc.)
```
Comment (jnh5y) [inline docs/user/spark.rst]: @tkunicki @aheyne

File: docs/user/spark.rst [hunk 6]
```diff
@@ -79,8 +257,8 @@ And create the RDD
     val rdd = GeoMesaSpark.rdd(new Configuration, sc, params, query)
 
 
-Examples
---------
+Further Examples
```
Comment (jnh5y) [inline docs/user/spark.rst]: Add section about Python RDDs @tkunicki

**Comment by mdzimmerman** [comment]
Replaced by #1324.

**Pull request closed without merging**